storm-core/src/jvm/backtype/storm/scheduler/Cluster.java

```diff
@@ -430,12 +457,44 @@ public SupervisorDetails getSupervisorById(String nodeId) {
     }
 
     /**
+     * set assignments for cluster
+     */
+    public void setAssignments(Map<String, SchedulerAssignment> newAssignments) {
+        this.assignments = new HashMap<String, SchedulerAssignmentImpl>();
+        for (Map.Entry<String, SchedulerAssignment> entry : newAssignments.entrySet()) {
```

Comment:
ok but here I am just creating a new map will the same structure as the map pass in as a parameter and just setting it.